stable/elasticsearch/templates/configmap.yaml

```diff
@@ -65,17 +67,37 @@ data:
         ping.unicast.hosts: ${DISCOVERY_SERVICE:}
         minimum_master_nodes: ${MINIMUM_MASTER_NODES:2}
 
-{{- if and ( .Values.cluster.xpackEnable ) ( gt $minorAppVersion 6.3 )  }}
+  {{- if and ( .Values.cluster.xpackEnable ) ( gt $minorAppVersion 6.3 )  }}
     # see https://www.elastic.co/guide/en/x-pack/current/xpack-settings.html
     # After 6.3 xpack systems changed and are enabled by default and different configs manage them this enables monitoring
     xpack.monitoring.collection.enabled: ${XPACK_MONITORING_ENABLED:false}
-{{- else if .Values.cluster.xpackEnable }}
+  {{- else if .Values.cluster.xpackEnable }}
     # see https://www.elastic.co/guide/en/x-pack/current/xpack-settings.html
     xpack.ml.enabled: ${XPACK_ML_ENABLED:false}
     xpack.monitoring.enabled: ${XPACK_MONITORING_ENABLED:false}
     xpack.security.enabled: ${XPACK_SECURITY_ENABLED:false}
     xpack.watcher.enabled: ${XPACK_WATCHER_ENABLED:false}
-{{- end }}
+  {{- end }}
+{{- else if hasPrefix "7." .Values.appVersion }}
+    # see https://github.com/kubernetes/kubernetes/issues/3595
+    bootstrap.memory_lock: ${BOOTSTRAP_MEMORY_LOCK:false}
+
+    discovery.seed_hosts:
+      - ${DISCOVERY_SERVICE:}
+    cluster.initial_master_nodes:
+      {{- $replicas := .Values.master.replicas | int }}
+      {{- range $i := untilStep 0 $replicas 1 }}
+      - {{ template "elasticsearch.master.fullname" $ }}-{{ $i }}
+      {{- end }}
+
+  {{- if .Values.cluster.xpackEnable }}
```

Comment:
Dead code.

Reply:
What do you mean it's dead?

Reply:
xpack config with oss images creates startup failure

Reply:
Sorry, misread it.
I nonetheless think this is a lot of duplication, we should keep this factored.